pkg/reconciler/pipeline/dag/dag.go

```diff
@@ -163,9 +200,7 @@ func addLink(pt string, previousTask string, nodes map[string]*Node) error {
 		return fmt.Errorf("task %s depends on %s but %s wasn't present in Pipeline", pt, previousTask, previousTask)
 	}
 	next := nodes[pt]
-	if err := linkPipelineTasks(prev, next); err != nil {
```

Comment:
I had troubleshooted till this point, thanks for taking it further, appreciate your efforts 🙏